bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,425 @@
+                )
+                logger.info('Packaging lambda project')
+                lambda_package(
+                    lambda_project_dir, lambda_s3_bucket, deployment_path_prefix
+                )
+                logger.info('Deploying lambda project')
+                try:
+                    lambda_deploy(
+                        lambda_project_dir,
+                        stack_name=deployment_pb.namespace + '-' + deployment_pb.name,
+                    )
+                except Exception as e:
+                    _cleanup_s3_bucket(lambda_s3_bucket, aws_config.region)
+                    return ApplyDeploymentResponse(status=Status.INTERNAL(str(e)))
+
+            logger.info('Finish deployed lambda project, fetching latest status')
+            res_deployment_pb = Deployment(state=DeploymentState())
+            res_deployment_pb.CopyFrom(deployment_pb)
+            state = self.describe(res_deployment_pb, yatai_service).state
+            res_deployment_pb.state.CopyFrom(state)
+            return ApplyDeploymentResponse(
+                status=Status.OK(), deployment=res_deployment_pb
+            )
+
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
+
+    def delete(self, deployment_pb, yatai_service):
+        try:
+            logger.debug('Deleting AWS Lambda deployment')
+            state = self.describe(deployment_pb, yatai_service).state
+            if state.state != DeploymentState.RUNNING:
+                message = (
+                    'Failed to delete, no active deployment {name}. '
+                    'The current state is {state}'.format(
+                        name=deployment_pb.name,
+                        state=DeploymentState.State.Name(state.state),
+                    )
+                )
+                return DeleteDeploymentResponse(status=Status.ABORTED(message))
+
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+
+            cf_client = boto3.client('cloudformation', aws_config.region)
+            stack_name = deployment_pb.namespace + '-' + deployment_pb.name
+            cloud_formation_stack_result = cf_client.describe_stacks(
+                StackName='{ns}-{name}'.format(
+                    ns=deployment_pb.namespace, name=deployment_pb.name
+                )
+            )
+            stack_result = cloud_formation_stack_result.get('Stacks')[0]
+            outputs = stack_result['Outputs']
+            bucket_name = ''
+            for output in outputs:
+                if output['OutputKey'] == 'S3Bucket':
+                    bucket_name = output['OutputValue']
+            if bucket_name:
+                _cleanup_s3_bucket(bucket_name, aws_config.region)
+
+            logger.debug('Deleting Lambda function and related resources')
+            cf_client.delete_stack(StackName=stack_name)
```

Comment:
should we check the result of delete_stack?

Reply:
delete_stack returns None.  If it has error, it will be caught on the outer layer